runtime/runtime.go

```diff
@@ -22,6 +22,7 @@ import (
 )
 
 var logger = util.GetLogger("[runtime] ")
+var ExtraNS = map[string]eval.Ns{}
```

Comment:
This is unrelated.

Reply:
You want me to pull these into a seperate PR?

Reply:
Please see my comment above about Go API; if you are building an alternative runtime, consider duplicating some code from the runtime package.

Reply:
My intent is not to build an alternative runtime, it is to make it easy to add new namespaces that are implemented in go. Doing that in a way that does not involve just forking elvish means I need to at least be able to have the runtime register new go foncs in their own namespace.

Reply:
Sorry if I wasn't clear.

Where do you intend to add new namespaces implemented in Go?

* Are you going to add them in the Elvish codebase? Then the current approach of calling `ev.InstallModule` works well.

* Are going to add them outside the Elvish codebase? Then that is what I meant by "an alternative runtime". Doing so means that you will be inevitably relying on Elvish's Go API. I am not ready to commit to a stable Go API, so I don't intend to support such use cases.